### PR TITLE
Addes recursive `.value` read detection in nested functions

### PR DESCRIPTION
**Pull request by XantreDev**
#552

### CODE REVIEW

**Comment by XantreDev** [comment]
I've didn't changed `useSignals` implementation. Don't know why test fails

File: packages/react-transform/src/index.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {
 	template,
 } from "@babel/core";
 import { isModule, addNamed } from "@babel/helper-module-imports";
-import type { VisitNodeObject } from "@babel/traverse";
+import { type VisitNodeObject } from "@babel/traverse";
```
Comment (rschristian) [inline packages/react-transform/src/index.ts:10]: This bumps up the minimum supported TS version considerably, same issue as #451. No real need for it either

Comment (XantreDev) [reply packages/react-transform/src/index.ts:10]: Biome changed it for some reason

Comment (rschristian) [reply packages/react-transform/src/index.ts]: Hm, that sounds like a weird default. Fair enough though.

File: packages/react-transform/test/node/index.test.tsx [hunk 2]
```diff
@@ -553,6 +553,26 @@ describe("React Signals Babel Transform", () => {
 				experimental: { noTryFinally: true },
 			});
 		});
+
+		it("recursevely propogates `.value` reads to parent component", () => {
```
Comment (rschristian) [inline packages/react-transform/test/node/index.test.tsx:557]: nit:

```suggestion
		it("recursively propogates `.value` reads to parent component", () => {
```

**Comment by XantreDev** [comment]
@rschristian I understood why test fails

It transforms unmanaged test to managed test. Which is not working, because of it do not uses queueMicrotask

https://github.com/XantreDev/signals/blob/46a701f7dfe76f42c9a1fd8131eea670f41f1f22/packages/react/runtime/test/browser/useSignals.test.tsx#L495-L527

https://github.com/XantreDev/signals/blob/46a701f7dfe76f42c9a1fd8131eea670f41f1f22/packages/react/runtime/test/browser/useSignals.test.tsx#L456-L458

Should I add workaround to fix transformation behavior?

Should this file even be transformed?

https://github.com/XantreDev/signals/blob/46a701f7dfe76f42c9a1fd8131eea670f41f1f22/karma.conf.js#L157-L169

**Comment by rschristian** [comment]
I'm not familiar w/ the React transform (or its tests) at all -- I haven't/don't use it.

@andrewiggins is the best person to ask really, I couldn't say.

**Comment by XantreDev** [comment]
> @rschristian I understood why test fails
> 
> It transforms unmanaged test to managed test. Which is not working, because of it do not uses queueMicrotask
> 
> https://github.com/XantreDev/signals/blob/46a701f7dfe76f42c9a1fd8131eea670f41f1f22/packages/react/runtime/test/browser/useSignals.test.tsx#L495-L527
> 
> https://github.com/XantreDev/signals/blob/46a701f7dfe76f42c9a1fd8131eea670f41f1f22/packages/react/runtime/test/browser/useSignals.test.tsx#L456-L458
> 
> Should I add workaround to fix transformation behavior?
> 
> Should this file even be transformed?
> 
> https://github.com/XantreDev/signals/blob/46a701f7dfe76f42c9a1fd8131eea670f41f1f22/karma.conf.js#L157-L169

@andrewiggins What is your thoughts about it?

**Comment by JoviDeCroock** [comment]
#582 should have you covered on this one, will close this one out as superseded by that one

**Pull request closed without merging**